tests/integration/test_vxlan_small_isis_ingress.yml

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2024 Cisco Systems, Inc. and its affiliates
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# SPDX-License-Identifier: MIT
+
+---
+# This playbook runs tests agains the data under tests/integration/host_vars/fabric_full_small_isis_ingress
+
+# Instructions for using this playbook:
+#
+# 1. Make sure you run the tests/integration/test_update_model_data playbook first
+#
+# 3. Run the playbook with the following command:
+#    ansible-playbook -i tests/integration/test_inventory.yml tests/integration/test_vxlan_small_isis_ingress.yml
+
+# Reset Fabric
+- hosts: fabric_empty
+  gather_facts: no
+
+  roles:
+    - test_remove
+
+# Create and Deploy Fabric Configuration
```

Comment:
Instead of duplicating the main playbook for every fabric type, I suggest we try the following approach instead.

Create a single playbook where the `Create and Deploy Fabric Configuration` section takes an ansible variable.

```yaml
# Create and Deploy Fabric Configuration
- hosts: '{{ hostvars_fabric_name }}'
  gather_facts: no

  roles:
    - test_create
```

Then I believe you should be able to run it like this and pick the fabric type you want to validate

```bash
ansible-playbook -i hosts test_inventory.yml -e 'hostvars_fabric_name=fabric_full_small_isis_ingress'
```

Reply:
If this works we can update the main playbook with the list of fabric profiles we support

Reply:
@mikewiebe sounds good to me. I have created a generic playbook "test_vxlan.yml" with ansible variable to pass the fabric profile name, I tested and it works well. So I have updated the PR with removing all other playbooks related to different fabric profiles. I also added detailed comment having list of the supported profiles and Ansible command example to deploy different profiles. Please review and let me know your inputs.

  e.g. ansible-playbook -i test_inventory.yml test_vxlan.yml -e 'fabric_profile=fabric_full_small_isis_multicast_asm'